lib/OpenQA/Script/CloneJob.pm

```diff
@@ -204,14 +205,18 @@ sub get_deps ($job, $options, $job_type) {
     return ($chained // [], $directly_chained // [], $parallel // []);
 }
 
-sub handle_tx ($tx, $jobid, $job, $base_url, $options, $clone_map, $depth, $child_list) {
+sub handle_tx ($tx, $url_handler, $options, $jobs) {
     my $res = $tx->res;
     my $json = $res->json;
-    if (!$tx->error && ref $json eq 'HASH' && $json->{id}) {
-        my $clone_id = $clone_map->{$jobid} = $json->{id};
-        print "Created job #$clone_id: $job->{name} -> $base_url/t$clone_id\n";
-        clone_job($_, $options, $clone_map, $depth + 1, $clone_id) for @$child_list;    # clone children if needed
-        return $clone_id;
+    if (!$tx->error && ref $json eq 'HASH' && ref $json->{ids} eq 'HASH') {
+        my $cloned_jobs = $json->{ids};
+        my $base_url = openqa_baseurl($url_handler->{local_url});
+        for my $orig_job_id (keys %$cloned_jobs) {
+            my $orig_job = $jobs->{$orig_job_id};
+            my $cloned_job_id = $cloned_jobs->{$orig_job_id};
+            print "Created job #$cloned_job_id: $orig_job->{name} -> $base_url/t$cloned_job_id\n";
```

Comment:
This will still be the same message as in before, just renamed variables, right?

Reply:
That's the idea. Of course the ordering of the log messages differs. In the current version these messages are printed after each post call but with my change all of these messages are printed at the end (after the single post call for all jobs and thus after all asset-related messages).

For better machine-readability we might want to add a `--json-output` flag (which would simply print the JSON response from the API as-is). That might be useful for going forward with the investigation script.